instrumentation/opentelemetry-instrumentation-pyramid/src/opentelemetry/instrumentation/pyramid/callbacks.py

```diff
@@ -178,7 +183,9 @@ def trace_tween(request):
                 else:
                     activation.__exit__(None, None, None)
 
-                context.detach(request.environ.get(_ENVIRON_TOKEN))
+                request_environ = request.environ
+                if _ENVIRON_TOKEN in request_environ.keys():
+                    context.detach(request.environ.get(_ENVIRON_TOKEN))
```

Comment:
I'd just run `.get(_ENVIRON_TOKEN, None)` directly and check if the result is `None` or not before detaching. I think it'd be one less lookup.